packages/service-client/odsp-client/src/OdspAudience.ts

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { assert } from "@fluidframework/core-utils";
+import { ServiceAudience } from "@fluidframework/fluid-static";
+import { type IClient } from "@fluidframework/protocol-definitions";
+
+import { type OdspMember, type IOdspAudience, OdspUser } from "./interfaces";
+
+/**
+ * @alpha
+ */
+export class OdspAudience extends ServiceAudience<OdspMember> implements IOdspAudience {
+	protected createServiceMember(audienceMember: IClient): OdspMember {
+		const user = audienceMember.user as OdspUser;
+		assert(user?.name !== undefined, 'Provided user was not an "OdspUser".');
```

Comment:
It looks like `email` is also a required property of `OdspUser`. We should verify it exists here too.